pulsar-client-cpp/wireshark/pulsarDissector.cc

```diff
@@ -78,62 +78,139 @@ static int hf_pulsar_message_permits = -1;
 
 static int ett_pulsar = -1;
 
-const static int FRAME_SIZE_LEN = 4;
+const static gint FRAME_SIZE_LEN = 4;
```

Comment:
Why do you change `int` to `gint`?

Reply:
You can see `tvb_get_ntohl(tvb, offset)`, the offset type is `gint`.

Reply:
If you insisted on the type safety, you should notice that `FRAME_SIZE_LEN` is also used in

```c++
uint32_t offset = FRAME_SIZE_LEN;
```

and

```c++
static uint32_t get_pulsar_message_len(packet_info *pinfo _U_, tvbuff_t *tvb, int offset,
                                    void *data _U_) {
    uint32_t len = (uint32_t) tvb_get_ntohl(tvb, offset);
    return FRAME_SIZE_LEN + len;
}
```

The type of `FRAME_SIZE_LEN` should be `uint32_t` actually.

Instead of changing the type, it's better to add the type cast explicitly to declare the type cast is safe. For example,

```c++
    uint32_t cmdSize = (uint32_t) tvb_get_ntohl(tvb, static_cast<gint>(offset));
```

However, it's not a big issue if you keep the current code. It's just something about the code quality.

Reply:
Thanks for your reminder, I changed it back.